src/index.ts

```diff
@@ -286,7 +286,9 @@ class Poll<T extends PromiseResultTypes> {
 		const index = subscribers.push(fn) - 1;
 
 		return {
-			unsubscribe: () => delete this.subscribers[name][index],
+			unsubscribe: () => {
+				this.subscribers[name].splice(index, 1);
```

Comment:
Delete from an array is unsafe as it does not modify the `.length` property:

See: https://typescript-eslint.io/rules/no-array-delete/